project/src/main/java/com/google/autograder/servlets/helpers/API.java

```diff
@@ -26,26 +33,36 @@
 
 public final class API {
 
-    private static final String RESOURCES_PATH = "../../../project/src/main/resources"; // "../classes"
+    private static final String RESOURCES_PATH  = "../../../project/src/main/resources"; // "../classes"
+    private static final Logger LOGGER = Logger.getLogger(API.class.getName());
     private static final String API_KEY_PATH = "/auth/api_key.json";
     private static final Gson GSON = new Gson();
 
-    public static final String API_KEY = getAPIKey();
     public static final String UTF_8 = StandardCharsets.UTF_8.name();
+    public static final String API_KEY;
 
-    private static String getAPIKey() {
-        String api_key = null;
+    static {
+        API_KEY = retrieveAPIKey().orElse(null);
+    }
+
+    private static Optional<String> retrieveAPIKey() {
+        String apiKeyPath = RESOURCES_PATH + API_KEY_PATH;
+        String apiKey = null;
 
         try {
-            String json = new String(Files.readAllBytes(Paths.get(RESOURCES_PATH + API_KEY_PATH)));
+            String json = new String(Files.readAllBytes(Paths.get(apiKeyPath)));
             JSONObject jsonObject = (JSONObject) new JSONParser().parse(json);
             JSONObject apiKeyObject = (JSONObject) new JSONParser().parse(jsonObject.get("api_key").toString());
-            api_key = apiKeyObject.get("api_key").toString();
-        } catch(Exception e) {
-            e.printStackTrace();
+            apiKey = apiKeyObject.get("api_key").toString();
```

Comment:
You can `return Optional.of(apiKeyObject.get("api_key").toString());` here rather than saving it to a local variable

Reply:
+1